presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -38,6 +38,22 @@ private MathFunctions() {}
 
     @Description("absolute value")
     @ScalarFunction("abs")
+    @SqlType(StandardTypes.TINYINT)
+    public static long absTinyInt(@SqlType(StandardTypes.TINYINT) long num)
+    {
+        return Math.abs(num);
```

Comment:
This is going to fail for `Byte.MIN_VALUE` as it will become `Byte.MAX_VALUE + 1`. Also, it looks like the implementation for `Integer` is broken as well